### PR TITLE
Add availabilityStatus enum to BoardingPass

### PR DESCRIPTION
**Pull request by Radek-Palisa**

### CODE REVIEW

File: src/booking/types/outputs/BoardingPass.js [hunk 1]
```diff
@@ -26,7 +29,12 @@ export default new GraphQLObjectType({
       type: GraphQLDate,
       description:
         'The date when the boarding pass will be available for download',
-      reslove: ({ availableAt }) => availableAt,
+      resolve: ({ availableAt }) => availableAt,
+    },
+    availabilityStatus: {
+      type: BoardingPassStatus,
+      description: 'Boarding pass availability status',
+      resolve: boardingPassStatusResolver(),
```
Comment (michalsanger) [inline src/booking/types/outputs/BoardingPass.js]: I see one issue here. If the resolver doesn't get what's expected there is no Flow check to let us know about it. So it's safer to use it like this:
https://github.com/kiwicom/graphql/blob/e1488b2c38cc4d3d26b93336d43027c23310b848/src/hotel/types/outputs/HotelCity.js#L38

File: src/booking/types/enums/BoardingPassStatus.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+export const boardingPassStatusResolver = () => ({
+  availableAt,
+  boardingPassUrl,
+}: ResolverType): string => {
+  if (boardingPassUrl) {
+    return AVAILABLE;
+  }
+
+  if (availableAt === null) {
```
Comment (michalsanger) [inline src/booking/types/enums/BoardingPassStatus.js]: What if `availableAt` is `undefined`?

File: src/booking/types/outputs/__tests__/BoardingPass.test.js [hunk 3]
```diff
@@ -57,4 +17,40 @@ Object {
...
   });
+
+  it('should return the corresponding availability status', () => {
+    const offsetDaysFromToday = offsetDays => {
+      return new Date(new Date().setDate(new Date().getDate() + offsetDays))
+        .toISOString()
+        .split('T')[0];
+    };
+
+    expect(
```
Comment (michalsanger) [inline src/booking/types/outputs/__tests__/BoardingPass.test.js]: It's better for readability to create it's own test for every expect.

File: src/booking/types/enums/BoardingPassStatus.js (hunk 2 above)
Comment (Radek-Palisa) [reply src/booking/types/enums/BoardingPassStatus.js]: If I understood the specs correctly, `null` will be set deliberately to the field when the boarding pass won't be available online and it can be obtained at the airport only. So `undefined` should only happen if there's a server error I suppose? In that case it will return `OTHER`.

Comment (michalsanger) [reply src/booking/types/enums/BoardingPassStatus.js]: OK if it's intentional. 

This should be covered in the unit tests:
> So undefined should only happen if there's a server error I suppose? In that case it will return OTHER.

File: src/booking/types/outputs/BoardingPass.js (hunk 1 above)
Comment (jaroslav-kubicek) [inline src/booking/types/outputs/BoardingPass.js]: there is probably also no need to define `boardingPassStatusResolver` as function and then calling it :)

**Comment by jaroslav-kubicek** [comment]
pls also rebase, cc @Radek-Palisa

File: src/booking/types/enums/BoardingPassStatus.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import { GraphQLEnumType } from 'graphql';
+import { DateTime } from 'luxon';
+
+const now = DateTime.utc();
```
Comment (tbergq) [inline src/booking/types/enums/BoardingPassStatus.js]: This means that now will be at the time the file starts to exist (code deployed) and not the time when the function is called.

File: src/booking/types/outputs/__tests__/BoardingPass.test.js [hunk 5]
```diff
@@ -1,52 +1,16 @@
...
 import BoardingPass from '../BoardingPass';
 import { evaluateResolver } from '../../../../common/services/TestingTools';
 
+const now = DateTime.utc();
+
+const availabilityStatus = BoardingPass.getFields().availabilityStatus;
+
 describe('Boarding pass', () => {
   it('should have valid fields', () => {
-    expect(BoardingPass.getFields()).toMatchInlineSnapshot(`
```
Comment (tbergq) [inline src/booking/types/outputs/__tests__/BoardingPass.test.js]: also why was this changed from inline snapshot, to snapshot?

Comment (Radek-Palisa) [reply src/booking/types/outputs/__tests__/BoardingPass.test.js]: I did this on the instruction of @jaroslav-kubicek. I assume the reason being that it's better to have the snapshot as a seperate file within the snapshot folder rather having the inlined in the test file.

File: src/booking/types/enums/BoardingPassStatus.js (hunk 4 above)
Comment (Radek-Palisa) [reply src/booking/types/enums/BoardingPassStatus.js]: thanks, that was silly.

**Comment by Radek-Palisa** [comment]
@michalsanger @jaroslav-kubicek @tbergq can you guys have a look? :)

**Comment by tbergq** [comment]
LGTM, please also rebase master into this branch since `This branch is out-of-date with the base branch`
